superset-frontend/plugins/plugin-chart-echarts/src/types.ts

```diff
@@ -168,4 +171,21 @@ export interface TreePathInfo {
   value: number | number[];
 }
 
+export class EchartsChartPlugin<
+  T extends SqlaFormData = SqlaFormData,
+  P extends ChartProps = ChartProps,
+> extends ChartPlugin<T, P> {
+  constructor(props: any) {
+    const { metadata } = props;
+    const mergedProps = {
+      ...props,
+      metadata: new ChartMetadata({
+        parseMethod: 'json',
+        ...metadata,
+      }),
+    };
+    super(mergedProps);
```

Comment:
nit: Maybe we don't need to create the intermediate `mergedProps`, but could just inline it in `super()`